src/main/java/com/feedzai/commons/sql/abstraction/batch/AbstractBatch.java

```diff
@@ -385,6 +385,7 @@ protected void start() {
     @Override
     public void close() throws Exception {
         destroy();
+        this.metricsListener.close();
```

Comment:
I am not sure if that is necessary, but consider if it would be necessary to check if `metricsListener` is not null.

Reply:
It shouldn't be.
I annotated the `BatchConfig.getMetricsListener()` with `@Nonnull`, meaning implementations should have something there; in the case of the `AbstractBatchConfig` with the current implementations extend, if the listener in the builder is null, then the final config will use `NoopMetricsListener`.
Same happens in the AbstractBatch deprecated constructors that have all separate parameters instead of BatchConfig.